ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml

```diff
@@ -388,42 +388,32 @@ objects:
           echo "Gathering artifacts ..."
           mkdir -p /tmp/artifacts/pods /tmp/artifacts/nodes /tmp/artifacts/metrics /tmp/artifacts/bootstrap /tmp/artifacts/network
 
-
-          if [ -f /tmp/artifacts/installer/terraform.tfstate ]
+          export PREFIX=$(grep cluster_id /tmp/artifacts/installer/terraform.tfvars | cut -d '"' -f4)
+          bootstrap_ip=$(openstack floating ip list --long -f csv -c Description -c "Floating IP Address" | grep "$PREFIX-bootstrap-fip" | cut -f1 -d "," | sed -e "s|\"||g")
+          if [ -n "${bootstrap_ip}" ]
           then
-              # TODO(trown): rework this to fetch bootstrap logs from openstack
-              # we don't have jq, so the python equivalent of
-              # jq '.modules[].resources."aws_instance.bootstrap".primary.attributes."public_ip" | select(.)'
-                #              bootstrap_ip=$(python -c \
-                #                  'import sys, json; d=reduce(lambda x,y: dict(x.items() + y.items()), map(lambda x: x["resources"], json.load(sys.stdin)["modules"])); k="aws_instance.bootstrap"; print d[k]["primary"]["attributes"]["public_ip"] if k in d else ""' \
-                #                  < /tmp/artifacts/installer/terraform.tfstate
-                #              )
-                #
-              if [ -n "${bootstrap_ip}" ]
```

Comment:
The previous setup was copy-pasted from installer-templates for 4.1 and is kept around as those templates are used for 4.1 and 4.2 CI clusters. since openstack is a 4.2 platform, I think it should be using the installer's gathering mechanism for bootstrapping failures.

Reply:
In the short term I think this gets us logs we desire for CI debugging.

Our CI environment requires a floating IP in order to access the bootstrap machine. The code in the current openshift-installer does not yet support detection of a floating IP. It tries to obtain the "local" (access_ipv4) address of the openstack server and thus fails to obtain logs in CI.

My goal here was to get good bootstrap logs in the short mid-term. Then we can look at the automatic openshift-installer feature to obtain logs as a separate issue.

NOTE: the code I'm fixing here exists in the templates for other CI jobs for AWS, etc. So if we aren't adding it here should we remove it from those CI job templates as well? See here: https://github.com/openshift/release/blob/master/ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml#L884

Reply:
Also, OpenStack floating IPs aren't a hard requirement. Some installations may use them, and some may not right? It just so happens that our CI environment requires them in order to obtain the logs at this point.

Reply:
> The previous setup was copy-pasted from installer-templates for 4.1 and is kept around as those templates are used for 4.1 and 4.2 CI clusters. since openstack is a 4.2 platform, I think it should be using the installer's gathering mechanism for bootstrapping failures.

I just pushed this to ci-tools which may help clarify to new developers this is not desired going forward: https://github.com/openshift/ci-tools/pull/82